src/Stack/Ghci.hs

```diff
@@ -836,13 +843,14 @@ checkForDuplicateModules pkgs = do
       pretty fp <+> parens (fillSep (punctuate "," (map displayPkgComponent (S.toList comps))))
 
 targetWarnings
-  :: HasRunner env
+  :: HasEnvConfig env
   => Path Abs File
   -> [(PackageName, (Path Abs File, Target))]
   -> [PackageName]
   -> Maybe (Map PackageName [Path Abs File], [Path Abs File])
+  -> Maybe (Project, Path Abs File)
   -> RIO env ()
-targetWarnings stackYaml localTargets nonLocalTargets mfileTargets = do
+targetWarnings stackYaml localTargets nonLocalTargets mfileTargets mproject = do
```

Comment:
Since the context has been changed from `HasRunner` to `HasEnvConfig`, there is no need to pass in `mproject`, instead `view $ configL.to configMaybeProject` can be done in this function.  Similarly, no need to pass in `stackYaml`, that can also be queried from the reader environment.

Reply:
Makes sense, done.